static/app/components/events/featureFlags/eventFeatureFlagList.tsx

```diff
@@ -0,0 +1,205 @@
+import {useCallback, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import {CompactSelect} from 'sentry/components/compactSelect';
+import DropdownButton from 'sentry/components/dropdownButton';
+import ErrorBoundary from 'sentry/components/errorBoundary';
+import {EventDataSection} from 'sentry/components/events/eventDataSection';
+import {FeatureFlagDrawer} from 'sentry/components/events/featureFlags/featureFlagDrawer';
+import useDrawer from 'sentry/components/globalDrawer';
+import KeyValueData, {
+  type KeyValueDataContentProps,
+} from 'sentry/components/keyValueData';
+import {IconChevron, IconSort} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {Event, FeatureFlag} from 'sentry/types/event';
+import type {Group} from 'sentry/types/group';
+import type {Project} from 'sentry/types/project';
+import useOrganization from 'sentry/utils/useOrganization';
+
+export enum FlagSort {
+  EVAL = 'eval',
+  ALPHA = 'alphabetical',
+}
+
+export const getLabel = (sort: string) => {
+  return sort === FlagSort.EVAL ? t('Evaluation Order') : t('Alphabetical');
+};
+
+export const FLAG_SORT_OPTIONS = [
+  {
+    label: getLabel(FlagSort.EVAL),
+    value: FlagSort.EVAL,
+  },
+  {
+    label: getLabel(FlagSort.ALPHA),
+    value: FlagSort.ALPHA,
+  },
+];
+
+export function EventFeatureFlagList({
+  event,
+  group,
+  project,
+}: {
+  event: Event;
+  group: Group;
+  project: Project;
+}) {
+  const [isCollapsed, setIsCollapsed] = useState(false);
+  const [sortMethod, setSortMethod] = useState<FlagSort>(FlagSort.EVAL);
+  const {closeDrawer, isDrawerOpen, openDrawer} = useDrawer();
+  const viewAllButtonRef = useRef<HTMLButtonElement>(null);
+
+  // transform the flags array into something readable by the key-value component
+  const hydrateFlags = (flags: FeatureFlag[]) => {
+    return flags.map(f => {
+      return {item: {key: f.flag, subject: f.flag, value: f.result}};
+    });
+  };
+
+  // TODO: remove this chunk --
+  const organization = useOrganization();
+  const flagsMap = organization.features.map(f => {
+    return {flag: f, result: true};
+  });
+  const initialFlags = hydrateFlags(flagsMap);
+  //----------------------------
+
+  // const initialFlags = hydrateFlags(event.contexts?.flags.values);
+  const [flags, setFlags] = useState<KeyValueDataContentProps[]>(initialFlags);
+
+  const handleSortEval = () => {
+    setFlags(initialFlags);
+  };
+
+  const handleSortAlphabetical = () => {
+    setFlags(
+      [...flags].sort((a, b) => {
+        return a.item.key.localeCompare(b.item.key);
+      })
+    );
+  };
+
+  const onViewAllFlags = useCallback(() => {
+    // good spot to track analytics
```

Comment:
added these comments so i can remember where to put analytics if we want them? -- can add these in now or in a future PR